### DISCUSSION TITLE
Update only "untracked" items

### DISCUSSION THREAD

**Discussion by tmllull** [tcgoetz/GarminDB#191] [General]
Hi! I found this amazing project a few days ago, and today I started to investigate it. After some tests, I have a doubt about the following point in section "Using it":

> Incrementally update your db by downloading the latest data and importing it by running garmindb_cli.py --all --download --import --analyze --latest in a terminal.

After first iteration finished (and maybe more than 2 hours), I test the "incremental" command. Although some processes are quicker than the first time, the processes "Processing FIT data from /HealthData/FitFiles/Monitoring" and "Processing FIT data from /HealthData/FitFiles/Activities" seems that is analyzing again all the data.

Is this correct? To only get the new data, is necessary to pass the "--all" flag?

Thanks for your help.

Best regards!

**Comment by tcgoetz** [comment]
The code determines what data to process based on the downloaded date. So if you immediately rerun after the initial download, it will reprocess a all of that data.

**Reply by tmllull** [reply]
Thanks for the response!

So, how should this "update" be used? I mean, If I run now a fresh execution, when this "update" only process the new data, only if a new activity exists?

EDIT: can I run the command without the "--all" flag after the first execution?

In addition, I notice that some information is downloaded only until yesterday, like sleep data. Today I have done a training, and the activity data has been downloaded, but the sleep data only have information until yesterday, is this a bug, or I need to configure something more?

**Reply by tcgoetz** [reply]
Updates with --latest work when the update is run in subsequent days. Some data is complete as soon as it appears and can be downloaded whenever it is visible in the API. Other types of data are complete only when the day is complete. It is easier to download that data only after the day its for is complete and to only process it then.
